packages/eslint-plugin/lib/tools.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-env node */
+
+'use strict';
+
+const isFunctionLike = node =>
+  [
+    'ArrowFunctionExpression',
+    'FunctionExpression',
+
+    'FunctionDeclaration',
+    'MethodDefinition',
+
+    'Program',
+    'StaticBlock',
+  ].includes(node.type);
+
+const isAwaitAllowedInNode = (node, already = new WeakSet()) => {
```

Comment:
mind adding a type signature? would be helpful to document the `node` type and to have an assurance that it always returns boolean.

Reply:
Definitely!